Hierarchies: Add a learning page for hierarchy definitions (#679)
iTwin/presentation: packages/hierarchies/learning/HierarchyProcessing.md

# Hierarchy processing

This page describes how and in what order a hierarchy provider (see `createHierarchyProvider`) processes nodes.

1. The hierarchy definition's `defineHierarchyLevel` function is called to get the hierarchy level definitions.
2. Hierarchy level definitions are turned into nodes.

   2.1. In case the hierarchy level definition is a `CustomHierarchyNodeDefinition`, it's simply mapped to the custom node.

   2.2. In case the hierarchy level definition is an `InstanceNodesQueryDefinition`, the nodes are fetched from the iModel and parsed either by definition's `parseNode` function, or the default parser.

3. Nodes are passed through the labels formatter.
4. Nodes are passed through the hierarchy definition's `preProcessNode` function, if one is defined.
5. Nodes are passed through the hiding processor that hides based on their `hideIfNoChildren` and `hideInHierarchy` processing flags. This step may require loading children for some nodes, e.g. to find out if they have children or get the children to replace the processed node.
6. Nodes are passed through the grouping processor.

> 7. Nodes' children flag is determined, if necessary. Unless the node already knows if it has children, the children are loaded to determine if the parent has children.

Reviewer: I think this could be said in a more concise manner. Something like this:
```suggestion
7. If nodes' children flag is undefined, it is determined by loading them.
```

Author: I like the idea to shorted what I had, but I'd like to reword a bit. What do you think about the following?

_7. If nodes' children flag is undefined, it is determined by loading the children._

Reviewer: That's also ok

Author: pushed
